### PR TITLE
Fire Extinguisher action "Toggle Safety"

### PR DESCRIPTION
**Pull request by alwinnocom**
image
image

### CODE REVIEW

File: Content.Server/Extinguisher/FireExtinguisherComponent.cs [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 [NetworkedComponent, RegisterComponent]
 [Access(typeof(FireExtinguisherSystem))]
 public sealed partial class FireExtinguisherComponent : SharedFireExtinguisherComponent
 {
+    /// <summary>
+    /// "ActionToggleSafety" refers to an entity created in Resources/Prototypes/Actions/extinguisher.yml
+    /// This configures the visual appearance of the button, including text description & sprites.
+    /// </summary>
+    [DataField("toggleAction", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string ToggleAction = "ActionToggleSafety";
```
Comment (deltanedas) [inline Content.Server/Extinguisher/FireExtinguisherComponent.cs:17]: replace with 
```
[DataField]
public EntProtoId ToggleAction = "ActionToggleSafety";`
```

File: Content.Server/Extinguisher/FireExtinguisherComponent.cs [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+
+    [DataField("toggleActionEntity")] public EntityUid? ToggleActionEntity;
```
Comment (deltanedas) [inline Content.Server/Extinguisher/FireExtinguisherComponent.cs:19]: just `[DataField]`

File: Content.Server/Extinguisher/FireExtinguisherSystem.cs [hunk 3]
```diff
@@ -19,8 +21,8 @@ public sealed class FireExtinguisherSystem : EntitySystem
     [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
     [Dependency] private readonly PopupSystem _popupSystem = default!;
     [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
-    [Dependency] private readonly SharedAudioSystem _audio = default!;
-
```
Comment (deltanedas) [inline Content.Server/Extinguisher/FireExtinguisherSystem.cs:23]: bruh

File: Content.Server/Extinguisher/FireExtinguisherSystem.cs [hunk 4]
```diff
@@ -30,13 +32,24 @@ public override void Initialize()
         SubscribeLocalEvent<FireExtinguisherComponent, AfterInteractEvent>(OnAfterInteract);
         SubscribeLocalEvent<FireExtinguisherComponent, GetVerbsEvent<InteractionVerb>>(OnGetInteractionVerbs);
         SubscribeLocalEvent<FireExtinguisherComponent, SprayAttemptEvent>(OnSprayAttempt);
+        /// <summary>
```
Comment (deltanedas) [inline Content.Server/Extinguisher/FireExtinguisherSystem.cs:35]: put xmldoc on the method itself not subscribing
also this has no reason to be documented the code is very simple

File: Content.Server/Extinguisher/FireExtinguisherSystem.cs [hunk 5]
```diff
@@ -30,13 +32,24 @@ public override void Initialize()
...
+        SubscribeLocalEvent<FireExtinguisherComponent, ToggleActionEvent>(OnToggleAction);
+        SubscribeLocalEvent<FireExtinguisherComponent, GetItemActionsEvent>(OnGetActions);
     }
 
     private void OnFireExtinguisherInit(EntityUid uid, FireExtinguisherComponent component, ComponentInit args)
     {
         if (component.HasSafety)
         {
             UpdateAppearance(uid, component);
+            /// <summary>
```
Comment (deltanedas) [inline Content.Server/Extinguisher/FireExtinguisherSystem.cs:48]: regular comment not xmldoc

File: Content.Server/Extinguisher/FireExtinguisherSystem.cs [hunk 6]
```diff
@@ -93,15 +106,23 @@ private void OnAfterInteract(EntityUid uid, FireExtinguisherComponent component,
                 uid, args.Target.Value);
         }
     }
-
+    /// <summary>
+    /// Uses ActionContainerSystem.cs to check if the entity is already an action.
+    /// Uses ActionEvents.cs to create the logic that registers the "ActionToggleSafety" entity in the Actions menu.  
+    /// </summary>
```
Comment (deltanedas) [inline Content.Server/Extinguisher/FireExtinguisherSystem.cs:112]: as above xmldoc unneeded its self explanatory

File: Content.Server/Extinguisher/FireExtinguisherSystem.cs [hunk 7]
```diff
@@ -130,14 +163,18 @@ private void OnSprayAttempt(EntityUid uid, FireExtinguisherComponent component,
         }
     }
 
-    public void ToggleSafety(EntityUid uid, EntityUid user,
-        FireExtinguisherComponent? extinguisher = null)
+    public void ToggleSafety(EntityUid uid,
```
Comment (deltanedas) [inline Content.Server/Extinguisher/FireExtinguisherSystem.cs:166]: user is never used just remove it

File: Content.Server/Extinguisher/FireExtinguisherSystem.cs [hunk 8]
```diff
@@ -130,14 +163,18 @@ private void OnSprayAttempt(EntityUid uid, FireExtinguisherComponent component,
...
+        FireExtinguisherComponent? extinguisher = null, EntityUid? user = null)
     {
         if (!Resolve(uid, ref extinguisher))
             return;
 
         extinguisher.Safety = !extinguisher.Safety;
         _audio.PlayPvs(extinguisher.SafetySound, uid, AudioParams.Default.WithVariation(0.125f).WithVolume(-4f));
         UpdateAppearance(uid, extinguisher);
+
+        // Change the sprite from Off to On or On to Off via SharedActionSystem.cs & ActionUIController.cs StartTargeting()
```
Comment (deltanedas) [inline Content.Server/Extinguisher/FireExtinguisherSystem.cs:176]: comment not needed probably

**Comment by Slava0135** [comment]
pro tip: don't use master for PRs - make a separate branch so you don't have to reset it and can work on several PRs at the same time

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Server/Extinguisher/FireExtinguisherSystem.cs (hunk 3 above)
Comment (Slava0135) [reply Content.Server/Extinguisher/FireExtinguisherSystem.cs:23]: :trollface:

Comment (alwinnocom) [reply Content.Server/Extinguisher/FireExtinguisherSystem.cs:23]: my bad it was an accident

**Comment by EmoGarbage404** [comment]
does this even need to be an action? toggling it inhand is literally just pressing "z," last I checked. This just seems like action bloat.

**Pull request closed without merging**